### PR TITLE
Divyanshu | Test | Bugs and fixes

### PR DESCRIPTION
**Pull request by dvCodeWorld**
https://app.clickup.com/t/86cugzwmg

### CODE REVIEW

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 1]
```diff
@@ -282,27 +295,43 @@ export class VatReportComponent implements OnInit, OnDestroy {
...
-     *
-     * @param {*} section
-     * @memberof VatReportComponent
-     */
+    * This will use for get year selected
+    *
+    * @param {*} selectedYear
+    * @memberof VatReportComponent
+    */
+    public getYearStartAndEndDate(selectedYear: any) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts]: add return type

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 2]
```diff
@@ -114,9 +122,12 @@ export class VatReportComponent implements OnInit, OnDestroy {
         private route: Router,
         private settingsBranchAction: SettingsBranchActions,
         private breakpointObserver: BreakpointObserver,
-        private modalService: BsModalService
+        private modalService: BsModalService,
+        private settingsFinancialYearActions: SettingsFinancialYearActions,
+        public settingsFinancialYearService: SettingsFinancialYearService
     ) {
-
+        this.store.dispatch(this.settingsFinancialYearActions.getFinancialYearLimits());
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts]: use of this store dispatch?

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 1 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts]: done

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts]: no use

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts [hunk 3]
```diff
@@ -409,4 +439,44 @@ export class VatReportComponent implements OnInit, OnDestroy {
...
+
+    /**
+    * Create financial years list for dropdown
+    *
+    * @private
+    * @param {*} startDate
+    * @param {*} endDate
+    * @memberof VatReportComponent
+    */
+    private createFinancialYearsList(startDate: any, endDate: any, initList: boolean = true): any {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/vat-report/vat-report.component.ts]: use of initList?

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/vat-report/vat-report.component.ts (hunk 3 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/vat-report/vat-report.component.ts]: Not  in use

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**